http/src/v0/block.rs

```diff
@@ -108,31 +112,176 @@ pub fn put<T: IpfsTypes>(
 }
 
+use futures::stream::Stream;
+use std::borrow::Cow;
+use std::convert::TryFrom;
+use std::fmt;
+
+fn rm_options() -> impl Filter<Extract = (RmOptions,), Error = Rejection> + Clone {
+    warp::filters::query::raw().and_then(|q: String| {
+        let res = RmOptions::try_from(q.as_str())
+            .map_err(StringError::from)
+            .map_err(warp::reject::custom);
+
+        futures::future::ready(res)
+    })
+}
+
+#[derive(Debug)]
+pub enum ParseError<'a> {
+    DuplicateField(Cow<'a, str>),
+    MissingArg,
+    InvalidNumber(Cow<'a, str>, Cow<'a, str>),
+    InvalidBoolean(Cow<'a, str>, Cow<'a, str>),
+}
+
+impl<'a> fmt::Display for ParseError<'a> {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        use ParseError::*;
+        match *self {
+            DuplicateField(ref s) => write!(fmt, "field {:?} was duplicated", *s),
+            MissingArg => write!(fmt, "required field \"arg\" missing"),
+            InvalidNumber(ref k, ref v) => write!(fmt, "field {:?} invalid number: {:?}", *k, *v),
+            InvalidBoolean(ref k, ref v) => write!(fmt, "field {:?} invalid boolean: {:?}", *k, *v),
+        }
+    }
+}
+
+impl<'a> TryFrom<&'a str> for RmOptions {
+    type Error = ParseError<'a>;
+
+    fn try_from(q: &'a str) -> Result<Self, Self::Error> {
+        use ParseError::*;
+
+        let parse = url::form_urlencoded::parse(q.as_bytes());
+
+        let mut args = Vec::new();
+        let mut force = None;
+        let mut quiet = None;
+
+        for (key, value) in parse {
+            let target = match &*key {
+                "args" => {
+                    args.push(value.into_owned());
+                    continue;
+                }
+                "force" => &mut force,
+                "quiet" => &mut quiet,
+                _ => {
+                    // ignore unknown fields
+                    continue;
+                }
+            };
+
+            // common bool field handling
+            if target.is_none() {
+                match value.parse::<bool>() {
+                    Ok(value) => *target = Some(value),
+                    Err(_) => return Err(InvalidBoolean(key, value)),
+                }
+            } else {
+                return Err(DuplicateField(key));
+            }
+        }
+
+        Ok(RmOptions {
+            args,
+            force: force.unwrap_or(false),
+            quiet: force.unwrap_or(false),
+        })
+    }
+}
+
+async fn rm_block<T: IpfsTypes>(ipfs: Ipfs<T>, cid: Cid) -> Result<RmResponse, Error> {
+    match ipfs.remove_block(&cid).await {
+        Ok(()) => Ok(RmResponse {
+            hash: cid.to_string(),
+            error: "".to_string(),
+        }),
+        Err(e) => Ok(RmResponse {
+            hash: "".to_string(),
+            error: e.to_string(),
+        }),
+    }
+}
+
+async fn rm_blocks<T: IpfsTypes>(
+    ipfs: Ipfs<T>,
+    cids: Vec<Cid>,
+) -> Result<impl Stream<Item = Result<RmResponse, Error>> + Send, Error> {
```

Comment:
```suggestion
) -> FuturesOrdered<Item = Result<RmResponse, Error>> {
```
The result seems extra, as I can't see a way for this to error completly?

Reply:
Looks like this is not how you annotate a `FuturesOrdered`, getting the "associated type bindings are not allowed here